File: server/QCLab/Models/Category.cs
using System;
using System.Collections.Generic;

namespace QCLab.Models;

public partial class Category
{
    public long Id { get; set; }

    public string Name { get; set; } = null!;

    public string Code { get; set; } = null!;

    public string? Description { get; set; }

    public DateTime DateCreated { get; set; }

    public bool IsObsolete { get; set; }

    public DateTime? DateObsolete { get; set; }

    public string? CommentsObsolete { get; set; }

    public virtual ICollection<Reception> Receptions { get; set; } = new List<Reception>();

    public virtual ICollection<Test> Tests { get; set; } = new List<Test>();
}

File: packages/client_react/src/types/category.ts
export interface CategoryDto {
  id: number;
  name: string;
  code: string;
  description: string | null;
  isObsolete: boolean;
  dateCreated: string;
  dateObsolete: string | null;
  commentsObsolete: string | null;
}

export interface CreateCategoryDto {
  name: string;
  code: string;
  description?: string | null;
}

export interface UpdateCategoryDto {
  name: string;
  code: string;
  description?: string | null;
}

export interface CategoryTestDto {
  id: number;
  name: string;
  code: string;
  typeId: number;
  isArray: boolean;
  typeName: string;
}

export interface UpdateCategoryTestsDto {
  testIds: number[];
}
